registry/domain/services/service_publisher_domain_service.py

```diff
@@ -78,18 +78,23 @@ def publish_service_data_to_ipfs(self, service, environment):
         # publish assets
         service = self.publish_service_proto_to_ipfs(service)
         self.publish_assets(service)
+        service_metadata = self.get_service_metadata(service, environment)
+        service_metadata_filename = f"{METADATA_FILE_PATH}/{service.uuid}_service_metadata.json"
+        json_to_file(service_metadata, service_metadata_filename)
+        service.metadata_uri = METADATA_URI_PREFIX + self.publish_file_to_ipfs(service_metadata_filename)
+        return service
+
+    @staticmethod
+    def get_service_metadata(service, environment):
         service_metadata = service.to_metadata()
         if environment == EnvironmentType.TEST.value:
-            for group in service.groups:
-                service_metadata["groups"][0]["endpoints"] = group.test_endpoints
-
+            for group_number in range(len(service.groups)):
```

Comment:
looks complex. we should simplify it down.

Reply:
i dont think we will be able to iterate it in better way. if we will edit it before generating metadata than it will also change the main network service object

Reply:
for record in service_metadata["groups"]:
  record.update({"freecalls":  BLOCKCHAIN_TEST_ENV["free_calls"]})

Reply:
group no in `service.groups` will be different from the service metadata group no